backend/s3/s3.go

```diff
@@ -2516,6 +2618,8 @@ func setQuirks(opt *Options) {
 		useMultipartEtag = false // untested
 	case "Wasabi":
 		// No quirks
+	case "Qiniu":
```

Comment:
Did you try running the integration tests against Qiniu to check the quirks are correct?

From source root

- `go build ./...`
- `test_all -remotes TestQiniu:`

Wait for result to appear in your browser.

The backend test is useful to run individually - it is the most likely one to fail, 

```
cd backend/s3
go test -v -remote TestQiniu:
```